### DISCUSSION TITLE
[Feature] <multi-account/profile switching support>

### DISCUSSION THREAD

**Discussion by monkeyotg** [immich-app/immich#2919] [Feature Request]
### The feature

Do you have any plans to add multi-account/profile switching support?

My Use Case 1:
I host Immich on my home server/computer. And to avoid losing the connection when I am out somewhere, I use Zerotier to create a Virtual LAN Connection. In this case, I even need to keep Zerotier running on my phone when I return home, back to the real LAN environment.

My Use Case 2:
I have 3 accounts set up for now.
One for my phone photos
One for my camera photos
One for my reference photo collection
Now I use the account for phone photos as the main account and access the pictures from the other two accounts via sharing. But in reality, I don't want these accounts to be associated in any way. It makes the list under the Sharing tab way too long.

So if with a multi-account/profile switching feature, it would be very helpful.

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by etnoy** [comment]
There are no current plans, although I personally would like the same features. Just like you say, it would be useful to have several independent libraries of photos that have different uses. Especially my boxes and boxes of scanned slides and negatives that can't be timestamped in a useful way

**Reply by monkeyotg** [reply]
Totally understandable, I mean,
it's a "better to have" feature. not something essential.
and your boxes and boxes of scanned slides and negatives.

But anyway Immich did a pretty good job! Glad to have such a solution for the photo library.

**Comment by zdimension** [comment]
A simple account switcher like Google's would be really nice! 

![image](https://github.com/immich-app/immich/assets/4533568/fd35cea3-77e7-44ce-82da-18d0382f6979)

I have an account for my personal (phone) photos and an account for the family photos (cameras, scanned slides, ...).

**Comment by bo0tzz** [comment]
Fwiw while not "native"/built-in, this should be pretty easy to accomplish with an oauth server.

**Comment by RogerHartmann** [comment]
I have another use case to throug in:
- while at home, I use a direct connection to my immich server.
- while outside home, I have setup an nginx proxy

The proxy solution is slower compared to the direct connection (maybe because my thin client does not have that much computing power)

Ideally, this could be realized as a fall-back solution, so if 1 server/account is not reachable, the fall-back would be used.

Well, a little bit different use case to the ones described above.

**Comment by fakeboboliu** [comment]
If we can switch accounts from the client side quickly, it would resolve most of the issues in https://github.com/immich-app/immich/issues/12614 without too much complexity.

Something like Synology Photos's "shared space" can be easily archived by sharing an account.

Does it make this feature much more nice to have?

**Reply by mmomjian** [reply]
No, we have a different vision for Better Sharing that is far beyond easy account switching.

**Comment by eagle470** [comment]
It would be good to have this as I don't want my user account to be an admin account. Separation of duties and all. 

Beyond that, allowing management of child accounts would be nice, or being able to have read/write permissions to other users libraries. I would find the ability to share a library as r/w an acceptable solution.

Arguably, I would say the main point would be to be able to search through a library without have to see every single picture in the account. 

So maybe allowing for search on specific albums and then being able to select to see those photos on the "roll" or the explore page, or not, would be suitable.

I don't want my memes, expense receipts and family photos all rolled together.

**Reply by Handrail9** [reply]
+1 on the memes and family photos being separated. I'm a data hoarder and trying to search all thru all my memes for family photos is currently a chore.

**Reply by fakeboboliu** [reply]
It would be nice, but Immich's devs have "a great vision" that would be awesome and overcome all our stupid users' opinions.

I'm not feeling bad about sponsoring such a project with greater-than-user visions with $100 for now, but I won't recommend anyone to pay for Immich anymore.

**Comment by rijndael86** [comment]
What happened with this? Will it move forward?

In Android you could make do with any of the solutions to install multiple instances of one app, but on iOS you can only have one, the other choice is to use the web version on safari and not be able to see full resolution photos...

**Comment by probablyjassin** [comment]
+1 this would be a great feature

**Comment by Finke3** [comment]
I‘m also interested in this feature. I want so split my photos from my mirror less camera and the daily snapshots with my mobile phone. Two separate profiles / timelines would be great, so that they are not mixed but still easily accessible.

**Comment by KBrygger** [comment]
Multi server/account would be a very welcome feature. For me, to be able to test features, upgrades and functionality on my test server, before upgrading my prod server.

**Comment by marvil6** [comment]
I agree! It would be super useful for those of us with different family / personal / hobby / etc. accounts. I implemented it  (fully functional) in #24857, but it was rejected for being too niche, which is understandable.

You can clone my fork and use that, but since I want actual updates (lol), I created a workaround for web; which you can find [here](https://code.marvil.co/jorge/snippets/src/branch/main/immich-account-switcher). It's very very janky and I would love official support (and for the mobile app as well)!

**Reply by fakeboboliu** [reply]
I don't understand why having multiple accounts itself would be considered too niche, considering that self-hosting a photo service is already a too niche operation.
But thank you very much for your attempt: you tried it before I had the energy and decided to invest the work into completing such a PR.

**Comment by probablyjassin** [comment]
are we allowed to reopen this in a new discussion since it was closed as outdated?

**Reply by fakeboboliu** [reply]
Maybe not, the org members don't seem to like the idea.

Of course, the reason for close might also be they have already figured out a more elegant way to solve this problem.

**Reply by harryfiedify** [reply]
Hmm, yes, but which solution might that be?

**Reply by BreadJS** [reply]
Can we contribute to this Idea? Cause I will make this work if this needs to work. Cause I really need this.

**Comment by bo0tzz** [comment]
We decided end of last year that we don't want account switching in Immich (https://github.com/immich-app/immich/pull/24857). We're working on sharing improvements (#12614) that should help decrease the need for it.

**Reply by daniele-athome** [reply]
Just wanted to add my 2 cents - probably someone else already said it: the most useful use case for this feature would actually be on mobile, i.e. to allow connecting to multiple servers.

For web browsing, using separate browser containers will do.

**Comment by plexx-dev** [comment]
first of all massive thanks to the immich team for the amazing software. You guys rock!

i wanted to throw in another use case for the account switcher in the mobile app.
currently i have two separate immich instances:
one for family pics etc
and one shared friend group server where we dump our photos from trips, hangouts etc.

right now if i want to switch between the instances. i need to logout, change the url, log back in and rebuild the timeline (10min with 100k photos).

a quick switcher like in the discord, nextcloud or google mobile apps would make using immich in my use case so much smoother

(i cant put the family account and friend group account on one server because the friends group server is reachable from the internet and the family server is only locally on my nas and not open to the internet)

**Comment by amk221** [comment]
+1 for a switcher

* I have a cameraroll immich account
* periodically move worthy photos into the main 'sacred' immich account
 
 so at any one time, using the immich app, I can only see the cameraroll immich account, not my main one.

**Reply by marvil6** [reply]
I have the same use-case, plus another 2 accounts that I also use like this.

**Comment by borgqueenx** [comment]
also add my thumbs up to this idea.
Im a bit sad the dev's decided against this....even though its already build and ready to implement, and quite some users wish for this feature.
If people dont want to use it.....they can simply not use it?
Disappointing that it's not added.
